cogs/malware_protection.py

```diff
@@ -0,0 +1,50 @@
+from contextlib import suppress
+from os.path import splitext
+
+import aiohttp
+from discord import Color, Embed, Message, NotFound
+from discord.ext.commands import Bot, Cog
+
+from assets import AntiMalware
+
+from .utils import constants
+
+FILE_EMBED_DESCRIPTION = (
+    f"""
+    **Oh No!** Your message got zapped by our spam filter.
+    We currently don't allow `.txt` attachments or any source files, so here are some tips you can use: \n
+    • Try shortening your message, if it exceeds 2000 character limit
+    to fit within the character limit or use a pasting service (see below) \n
+    • If you're showing code, you can use codeblocks or use a pasting service like :
+    {constants.paste_link} or {constants.paste_link_2}
+    """
+)
+
+
+class MalwareProtection(Cog):
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+
+    @Cog.listener()
+    async def on_message(self, message: Message) -> None:
+        """Find messages with blacklisted attachments."""
+        if not message.attachments or not message.guild:
+            return
+        # TODO : change this to manage messages perms
+        elif message.author.guild_permissions.administrator:
+            return
+
+        file_extensions = {splitext(attachment.filename.lower())[1] for attachment in message.attachments}
+        is_blocked = file_extensions - set(AntiMalware.whitelist)
+
+        if is_blocked:
+            embed = Embed(description=FILE_EMBED_DESCRIPTION, color=Color.dark_blue())
+
+            with suppress(NotFound, ConnectionError):
+                await message.delete()
```

Comment:
You can work on this, or at least keep it here as TODO.

```suggestion
                # TODO: Adjust the message and remove only filtered attachments instead of deleting the whole message
                await message.delete()
```